src/Shared/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -117,6 +117,12 @@ public void Configure(TelemetryConfiguration configuration)
                     configuration.TelemetryChannel.EndpointAddress = this.applicationInsightsServiceOptions.EndpointAddress;
                 }
 
+                // Need to set connection string before calling Initialize() on the Modules and Processors.
+                if (this.applicationInsightsServiceOptions.ConnectionString != null)
+                {
+                    configuration.ConnectionString = this.applicationInsightsServiceOptions.ConnectionString;
```

Comment:
will this overwrite the channel.endpoint as well?

Reply:
Yes, which is why I put it last.